### ISSUE TITLE
Delete expired data in s3 bucket

### ISSUE THREAD

**Issue by pyhita** [grafana/pyroscope#3665]
Our team is using pyroscope to render flame graph information. We use s3 bucket as the back-end storage of pyroscope, but we need to regularly clean up the data in the s3 bucket that is older than 90 days. We have not seen that pyroscope provides APIs or commands to delete expired data.So we are wondering if we can delete it directly through the deletion mechanism of s3. We see that the storage format of pyroscope in s3 is as follows:

![Image](https://github.com/user-attachments/assets/dbf2bbc6-b0bd-4033-8c7d-64c3c04c40c8)
We plan to use aws s3 sdk to regularly delete objects older than 90 days in the grafana-pyroscope-data/anonymous/phlaredb directory to clean up expired data. We don’t know whether this will affect the running of pyroscope, so we want to confirm this. Is this cleaning method feasible?

**Comment by kolesnikovae** [comment]
You may also want to configure `-compactor.blocks-retention-period` ([link](https://grafana.com/docs/pyroscope/latest/configure-server/reference-configuration-parameters/)):
```yaml
# Delete blocks containing samples older than the specified retention period. 0
# to disable.
# CLI flag: -compactor.blocks-retention-period
[compactor_blocks_retention_period: <duration> | default = 0s]
```

However, it's a good idea is too configure retention policies in the object storage (but it should cover the compactor's config) – https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html

**Comment by pyhita** [comment]
Thank you for your information about compactor.blocks-retention-period, we have not noticed that pyroscope provides similar configuration parameters before.Configuring s3 bucket lifecycle requires additional permissions,so can we only configure this parameter to achieve the purpose of cleaning expired data?

**Comment by kolesnikovae** [comment]
Yes, this approach should generally be sufficient. However, in rare cases, some remnants, such as partially uploaded blocks, may still reside in object storage even after the retention policy has expired in the compactor

**Comment by pyhita** [comment]
Got it, thanks for your help!
